src/Simulator.py

```diff
@@ -184,6 +187,10 @@ def _handle_event(self,event):
                 random_node = np.random.choice(self._nodes)
                 random_node.receive_message()
 
+            case 'prepare_ballot_msg':
+                random_node = np.random.choice(self._nodes)
```

Comment:
should there not be a prioritization selection of nodes (also random, but relative to the priority)?

Reply:
In the event itself there is no priority as all nodes should take part in the protocol. In the nomination phase of the protocol there are priority nodes, but these are selected in the event itself, not prior. The balloting phase doesn't have priority nodes like the Nomination phase though.